src/controller/cap-level-controller.ts

```diff
@@ -158,7 +158,8 @@ class CapLevelController implements ComponentAPI {
         hls.autoLevelCapping = maxLevel;
         if (
           hls.autoLevelCapping > this.autoLevelCapping &&
-          this.streamController
+          this.streamController &&
+          hls.autoLevelCapping > this.streamController.currentLevel
```

Comment:
> User manually sets "level 5" quality.

I agree that manually a selected level should not be flushed.

 `this.streamController.currentLevel` could be from automatic selection and the player should remove the buffer if the currentLevel was arrived at automatically - to, for example, remove video that aliases when scaled down.

The issue here is that you do not want media removed when in manual mode. The fix here is to guard when the player is in manual mode:

```suggestion
          hls.autoLevelEnabled
```